src/field/FieldParamsPointers.hh

```diff
@@ -0,0 +1,80 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file FieldParamsPointers.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Macros.hh"
+#include "base/Types.hh"
+#include "base/Units.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Host and device parameters for propagation in a magnetic field and
+ * suggested default values
+ */
+struct FieldParamsPointers
+{
+    //! the minimum length of the field step
+    real_type minimum_step = 1.0e-5 * units::millimeter;
+
+    //! the closest miss distrance
+    real_type delta_chord = 0.25 * units::millimeter;
+
+    //! accuracy of intersection of the boundary crossing
+    real_type delta_intersection = 1.0e-4 * units::millimeter;
+
+    //! the relative error scale on the step length
+    real_type epsilon_step = 1.0e-5;
+
+    //! the maximum of the error ratio
+    real_type epsilon_rel_max = 1.0e-3;
+
+    //! the truncation error tolerance
+    real_type errcon = 1.0e-4;
+
+    //! pgrow (the exponent to increase a stepsiz)
+    real_type pgrow = -0.20;
+
+    //! pshrink (the exponent to decrease a stepsiz)
+    real_type pshrink = -0.25;
+
+    //! safety (a scale factor for the predicted step size)
+    real_type safety = 0.9;
+
+    //! the maximum scale to increase a step size
+    real_type max_stepping_increase = 5;
+
+    //! the maximum scale factor to decrease a step size
+    real_type max_stepping_decrease = 0.1;
+
+    //! the maximum number of steps (or trials)
+    size_type max_nsteps = 100;
+
+    //! Check whether the data is assigned
+    explicit inline CELER_FUNCTION operator bool() const
+    {
+        // clang-format off
+        return    minimum_step
+               && delta_chord 
+               && delta_intersection 
+               && epsilon_step 
+               && epsilon_rel_max 
+               && errcon 
+               && pgrow 
+               && pshrink 
+               && safety 
+               && max_stepping_increase
+               && max_stepping_decrease 
+               && max_nsteps;
```

Comment:
I think most of these conditions need to be more strict than `!= 0`— most need to be positive (some must be negative) and there should probably be some relationships between them as well such as  `minimum_step < delta_intersection`

Reply:
I was going to drop this function (as they are all defined with suggested default values), but added valid ranges with your suggestion.  There may be some relationships between them, but I would leave them independently until optimizations are completed - of course, all of these are from here and there from Geant4 codes, so some parameters are consolidated different way or removed later (once all field codes are integrated and optimized).  I will keep your suggestion in mind.

Reply:
That sounds like a good plan. Thanks!